jlm/llvm/opt/alias-analyses/Steensgaard.cpp

```diff
@@ -1379,45 +1379,69 @@ Steensgaard::AnalyzeMemcpy(const jlm::rvsdg::simple_node & node)
 {
   JLM_ASSERT(is<Memcpy>(&node));
 
-  // FIXME: handle unknown
-
-  // FIXME: write some documentation about the implementation
-
   auto & dstAddress = LocationSet_->Find(*node.input(0)->origin());
   auto & srcAddress = LocationSet_->Find(*node.input(1)->origin());
 
+  // Handle flags
+  //
+  // The dst address needs to have at least the same flags as the src address, because memcpy copies
```

Comment:
Are you sure about this? If I have code like
```c
int globalInt = 7;

void main() {
    int localInt;
    memcpy(&localInt, &globalInt, size of(int));
}
```

Then the register pointing to `localInt` should not be marked as PointingToEscaped, or any other flags for that matter?

I think we only need to do the "deeper" copy, and copy flags from srcMemory to dstMemory?

Reply:
You are right. Your example illustrates it quite well. What I did there did not lead to wrong results, but was too conservative. I changed it such that the flags get assigned to `srcMemory`. Please have a look again.